src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,175 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { useEffect, useRef, useState } from 'react';
+
+type Props = {
+  todo: Todo;
+  onDelete?: (todoId: number) => void;
+  onloadingTodoIds?: number[];
+  onEdit?: (currentTodo: Todo) => void;
+  isEditingTodo?: boolean;
+  setIsEditingTodo?: (isEditing: boolean) => void;
+  onChecked?: (currentTodo: Todo) => void;
+  errorFromServer?: boolean;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onloadingTodoIds = [],
+  onDelete = () => {},
+  onEdit = () => {},
+  isEditingTodo,
+  setIsEditingTodo = () => {},
```

Comment:
it's internal values related to the `TodoItem` only - remove this state the App.tsx and add here